src/extension/providers/nostr/index.ts

```diff
@@ -10,16 +12,23 @@ declare global {
 export default class NostrProvider {
   nip04 = new Nip04(this);
   enabled: boolean;
+  private _eventEmitter: EventEmitter;
 
   constructor() {
     this.enabled = false;
+    this._eventEmitter = new EventEmitter();
   }
 
   async enable() {
     if (this.enabled) {
-      return { enabled: true };
+      return Promise.resolve({ enabled: true });
     }
-    return await this.execute("enable");
+    return this.execute("enable").then((result) => {
```

Comment:
should this be awaited becuse of the async function?

Reply:
+1. I think mixing promises & non-promises (async/await) should be avoided because it makes it harder to read. 

Isn't that a lot easier to read? 

```
var result = await this.execute("enable");
if (typeof result.enabled === "boolean") {
  this.enabled = result.enabled;
}
return result;
```

Reply:
all the instances have been updated to this. resolved